circleguard/investigator.py

```diff
@@ -226,42 +236,137 @@ def median_frametime(frametimes):
         return np.median(frametimes)
 
     @staticmethod
-    def _parse_beatmap(beatmap):
+    def _parse_beatmap(beatmap, easy, hard_rock):
         hitobjs = []
 
         # parse hitobj
-        for hit in beatmap.hit_objects_no_spinners:
+        for hit in beatmap.hit_objects(easy=easy, hard_rock=hard_rock):
+            time = hit.time.total_seconds() * 1000
+            end_time = time
+            obj_type = 0
+            # object end time is different depending on the type of object
+            # type of hit object is necessary for notelock calculations (0 circle, 1 slider, 2 spinner)
+            if hasattr(hit, "end_time"):
```

Comment:
don't use attr checks ever if you can avoid it. Check against the instance of `hit` - `if isinstance(hit, Slider)`, etc